frb_codegen/src/source_graph.rs

```diff
@@ -209,34 +210,70 @@ impl Module {
+                    } else {
+                        for ident in idents {
+                            let ident_str = ident.to_string();
+                            scope_structs.push(Struct {
+                                ident,
+                                src: item_struct.clone(),
+                                visibility: syn_vis_to_visibility(&item_struct.vis),
+                                path: {
+                                    let mut path = self.module_path.clone();
+                                    path.push(ident_str);
+                                    path
+                                },
+                                mirror: true,
+                            });
+                        }
+                    }
                 }
                 syn::Item::Enum(item_enum) => {
-                    let (ident, mirror) = get_ident(&item_enum.ident, &item_enum.attrs);
-                    let ident_str = ident.to_string();
-                    scope_enums.push(Enum {
-                        ident,
-                        src: item_enum.clone(),
-                        visibility: syn_vis_to_visibility(&item_enum.vis),
-                        path: {
-                            let mut path = self.module_path.clone();
-                            path.push(ident_str);
-                            path
-                        },
-                        mirror,
-                    });
+                    let idents = get_ident(&item_enum.attrs);
+                    if idents.is_empty() {
+                        let ident = item_enum.ident.clone();
+                        let ident_str = ident.to_string();
+                        scope_enums.push(Enum {
+                            ident,
+                            src: item_enum.clone(),
+                            visibility: syn_vis_to_visibility(&item_enum.vis),
+                            path: {
+                                let mut path = self.module_path.clone();
+                                path.push(ident_str);
+                                path
+                            },
+                            mirror: false,
+                        });
+                    } else {
+                        for ident in idents {
+                            let ident_str = ident.to_string();
+                            scope_enums.push(Enum {
+                                ident,
+                                src: item_enum.clone(),
+                                visibility: syn_vis_to_visibility(&item_enum.vis),
+                                path: {
+                                    let mut path = self.module_path.clone();
+                                    path.push(ident_str);
+                                    path
+                                },
+                                mirror: true,
+                            });
+                        }
+                    }
```

Comment:
same comments as above. Btw would be even greater if can extract common code between enum and struct handling